src/Address/PostAddressFactory.php

```diff
@@ -6,25 +6,33 @@
 
 class PostAddressFactory
 {
+    /**
+     * @param null|string $country
+     * @param null|string $postalCode
+     * @param null|string $cityName
+     * @param null|string $address
+     * @return PostAddress
+     */
     public function createFlexible(?string $country, ?string $postalCode, ?string $cityName, ?string $address): PostAddress
     {
         return PostAddress::createForeignAddress($country, $postalCode, $cityName, $address);
     }
 
+    /**
+     * @param Address $address
+     * @return PostAddress
+     */
     public function createFromAddress(Address $address): PostAddress
     {
-        if ($address->isFrenchAddress()) {
-            return PostAddress::createFrenchAddress(
+        return $address->isFrenchAddress() ?
```

Comment:
The ternary operator is not suitable here as it makes the code less readable. This is why there's an explicit `if`  statement.